spec/views/user_show_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+
+RSpec.describe 'user show page', type: :feature do
+  before :each do
+    @user = User.create(name: 'Henry', photo: 'https://example.com/photo.jpg', bio: 'Lorem ipsum dolor sit amet')
+    @post = Post.create(user: @user, title: 'Post 1', text: 'Lorem ipsum dolor sit amet')
+    visit user_path(@user)
+  end
+
+  it 'can see the user\'s profile picture' do
+    expect(page).to have_css('img[src="https://example.com/photo.jpg"]')
+  end
+
+  it 'can see the user\'s name' do
+    expect(page).to have_content('Henry')
+  end
+
+  it 'can see the number of posts each user has made' do
+    expect(page).to have_content('1')
+  end
+
+  it 'can see the user\'s bio' do
+    expect(page).to have_content('Lorem ipsum dolor sit amet')
+  end
+
+  it 'can see the user\'s most recent three posts' do
+    expect(page).to have_content('Post 1')
+  end
+
+  it 'can see a button that lets me view all of the user\'s posts' do
+    expect(page).to have_link('Show all user post')
+  end
+
+  it 'when I click a user\'s post, I am redirected to the post show page' do
+    click_on 'Post 1'
+    expect(page).to have_current_path user_post_path(@user, @post)
+  end
+
+  it 'When I click to see all posts, I am redirected to the user\'s post index page' do
+    click_on 'Show all user post'
+    expect(page).to have_current_path user_posts_path(@user)
+  end
+end
```

Comment:
- [ ] Great job so far 👏🏻 however you should have implemented the featured test for **Login page**:

* Tests that returns a detailed error when no `username` and `password` is entered when the submit button is clicked
* Tests that returns a detailed error when an incorrect `username` or `password` is entered when the submit button is clicked
* Tests that redirects to the root page when a correct `username` and `password` is entered.

**See example code below**


```ruby
 describe 'the signin process', type: :feature do
     before :each do
         User.create(email: 'user1@gmail.com', password: 'password', name: 'User1')
     end

     it 'signs @user in' do
         visit '/users/sign_in'
         fill_in 'Email', with: 'user1@gmail.com'
         fill_in 'Password', with: 'password'
         click_button 'Log in'
         expect(current_path).to eq(root_path)
         expect(page).to have_text('Signed in successfully.')
     end

     it 'does not sign in with wrong info' do
         visit '/users/sign_in'
         fill_in 'Email', with: 'user1@gmail.com'
         fill_in 'Password', with: 'mypassword'
         click_button 'Log in'
         expect(page).to have_text('Invalid Email or password.')
     end
 end
```

**Note** Your code does not need to be exact.

Reply:
Thanks @BenMukebo  for the review. Though the requirements request tests for the login page, this has not been a previously requested part of the wireframe build and has yet to be implemented at this stage. Please reference this [**issue**](https://github.com/microverseinc/curriculum-rails/issues/107) that was submitted to correct this requirement for this project

Reply:
Hello Reviewer

Also, I did mention this on the pull request.

Reply:
Please check through the Readme file, all the authors are mentioned there

Reply:
@hello @TesheMaximillan, @dewslyse
As you can see on the screenshot below 👇🏻 I did not invent the project requirements but If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

Reply:
Hello @BenMukebo;

It's not about whether the code review is fair or not, Microverse is making changes to the curriculum, and we know we have to do the login page, actually, we didn't do the login page and it's not possible to make a test without having the login page.  Thank you for your understanding and please try to see the issue page too. [issue](https://github.com/microverseinc/curriculum-rails/issues/107)

Reply:
Hello @TesheMaximillan , your project is complete.